lib/valueflow.cpp

```diff
@@ -2647,6 +2647,8 @@ static void valueFlowLifetimeClassConstructor(Token* tok,
                 if (it == scope->varlist.cend())
                     return;
                 const Variable& var = *it;
+                if (var.valueType() && var.valueType()->container && var.valueType()->container->stdStringLike && !var.valueType()->container->view)
```

Comment:
I don't think you need to check stdLikeString. A non-view container should always have ownership. 

Looking at this closer it seems like this check should happen in the isLifetimeBorrowed function, but it probably doesn't get the parentValueType correctly.